File: backend/src/models/Conversation.ts
export interface DbConversationRow {
  id: string;
  type: 'direct' | 'group';
  group_id: string | null;
  direct_user_a_id: string | null;
  direct_user_b_id: string | null;
  created_at: Date;
}

/** A conversation summary as returned by the conversations-list endpoint —
 * includes the other participant (direct conversations only, V1's only
 * supported type) and the last message, so the Flutter list screen never
 * needs a second round-trip per row. */
export interface ConversationSummary {
  id: string;
  type: 'direct' | 'group';
  otherParticipant: { id: string; displayName: string | null } | null;
  lastMessage: {
    id: string;
    messageType: 'text' | 'location';
    body: string | null;
    senderUserId: string;
    createdAt: string;
  } | null;
  unreadCount: number;
  updatedAt: string;
}

File: lib/core/models/conversation.dart
import 'message.dart';

class ConversationLastMessage {
  final String id;
  final MessageType messageType;
  final String? body;
  final String senderUserId;
  final DateTime createdAt;

  const ConversationLastMessage({
    required this.id,
    required this.messageType,
    required this.body,
    required this.senderUserId,
    required this.createdAt,
  });

  factory ConversationLastMessage.fromJson(Map<String, dynamic> json) {
    return ConversationLastMessage(
      id: json['id'] as String,
      messageType: (json['messageType'] as String) == 'location' ? MessageType.location : MessageType.text,
      body: json['body'] as String?,
      senderUserId: json['senderUserId'] as String,
      createdAt: DateTime.parse(json['createdAt'] as String),
    );
  }
}

class ConversationSummary {
  final String id;
  final String? otherParticipantId;
  final String? otherParticipantName;
  final ConversationLastMessage? lastMessage;
  final int unreadCount;
  final DateTime updatedAt;

  const ConversationSummary({
    required this.id,
    required this.otherParticipantId,
    required this.otherParticipantName,
    required this.lastMessage,
    required this.unreadCount,
    required this.updatedAt,
  });

  factory ConversationSummary.fromJson(Map<String, dynamic> json) {
    final other = json['otherParticipant'] as Map<String, dynamic>?;
    final lastMessageJson = json['lastMessage'] as Map<String, dynamic>?;
    return ConversationSummary(
      id: json['id'] as String,
      otherParticipantId: other?['id'] as String?,
      otherParticipantName: other?['displayName'] as String?,
      lastMessage: lastMessageJson != null ? ConversationLastMessage.fromJson(lastMessageJson) : null,
      unreadCount: json['unreadCount'] as int? ?? 0,
      updatedAt: DateTime.parse(json['updatedAt'] as String),
    );
  }
}
